[Gutenberg 7.7.1]: E2E test fixes

ramonjd opened the pull request:
## Changes proposed in this Pull Request

Thanks for dropping by for what is tipped to be the most exciting PR of 2020^.

If things have gone to plan, we've fixed all the [E2E fails](https://github.com/Automattic/wp-calypso/pull/39292) introduced by [Gutenberg 7.7.1](https://github.com/Automattic/wp-calypso/issues/40078).

You won't be able disguise your awe as you scroll down the page and bask in the radiant glory of green checks in the merge checklist. 

Here, take some more! ✅✅✅✅✅✅✅✅

How does that feel?

_^ according to an exclusive poll of this PR's assignee list_

## Testing instructions

The **Needs e2e Testing Gutenberg** tests should all bear a ✅ 

## ⚠️ Please don't merge until 7.7.1 is in master

🙏 Thank you for your understanding.

ramonjd committed:
Updating paragraph selector

ramonjd committed:
Updating paragraph selector and unified variable name

ramonjd committed:
Editor post preview and dashicon classname updates

ramonjd committed:
Editor close publish pane dashicon classname updates

ramonjd committed:
Added external window page component for Gutenberg external window

ramonjd committed:
Added external window post component for Gutenberg external window

ramonjd committed:
Don't expect the #main element in new window until we've switched to ti

ramonjd committed:
Mobile views use the iframe preview pane

noahtallen commented:
> according to an exclusive poll of this PR's assignee list

no one is assigned 🤔 🤔 🤔

ramonjd commented:
> no one is assigned 🤔 🤔 🤔

Shhh, don't mess with the narrative :trollface:

bsessions85 reviewed (approved):
This looks good and I love all the ✅s! Thanks for the updates

ramonjd commented on test/e2e/lib/gutenberg/gutenberg-editor-component.js:
```diff
@@ -336,11 +336,21 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 
 	async launchPreview() {
-		return await driverHelper.clickWhenClickable(
+		if ( driverManager.currentScreenSize() === 'mobile' ) {
```
We might have to revert some of these changes if we don't end up using the device switcher preview button in 7.7.

The [media queries simulator magic](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/use-simulated-media-query/index.js#L18) won't work X-domain.

See: p7DVsv-8gl-p2#comment-27501

simison replied:
I commented out/simplified the desktop version but left this mobile `if` in place since we'll need later anyway.

simison commented:
7.7.1 is now merged to simple sites, atomic sites still in 7.3.0

ramonjd committed:
The `Use the Calypso Media Modal: @parallel` e2e test is failing on mobile and desktop because webdriver can't close the page template selector.

We're temporarily disabling this test until we smooth out the integration of Gutenberg 7.7.1 See: #40078

simison committed:
Disable "Basic Public Post" test in IE11 for now

mmtr committed:
Add extra timeout

simison committed:
Fix preview button in test spec

simison committed:
Fix preview button in test spec 2

simison committed:
Revert new Gutenberg Preview and use old Calypso preview instead

mmtr committed:
Perform "use blank" click with JS

ockham commented on test/e2e/specs/wp-calypso-gutenberg-post-editor-spec.js:
```diff
@@ -243,7 +244,8 @@ describe( `[${ host }] Calypso Gutenberg Editor: Posts (${ screenSize })`, funct
 	} );
 
-	describe( 'Basic Public Post @canary @ie11canary @parallel', function() {
+	// @TODO: add `@ie11canary` back here once FSE plugin works with IE11 again.
```
Can we add this back now that #40435 is merged? (Might need a rebase first.)

mmtr replied:
Not familiar with the FSE workflow nowadays, but I think we need D40802-code to be deployed first.

simison replied:
It'll need FSE plugin deploy to simple sites too — once it's done we can enable but it can be in separate PR.

simison replied:
Prepped a PR to enable https://github.com/Automattic/wp-calypso/pull/40443 — good to go once D40826-code is deployed.

mmtr committed:
Use JS click trick for visiting a published post/page
